src/main/kotlin/com/github/avrokotlin/avro4k/Avro.kt

```diff
@@ -15,266 +15,322 @@ import kotlinx.serialization.BinaryFormat
 import kotlinx.serialization.DeserializationStrategy
 import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.SerialFormat
+import kotlinx.serialization.SerializationException
 import kotlinx.serialization.SerializationStrategy
 import kotlinx.serialization.descriptors.SerialDescriptor
 import kotlinx.serialization.modules.SerializersModule
 import kotlinx.serialization.modules.contextual
+import kotlinx.serialization.serializer
+import org.apache.avro.AvroRuntimeException
 import org.apache.avro.Schema
 import org.apache.avro.file.CodecFactory
 import org.apache.avro.generic.GenericRecord
-import java.io.*
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.File
+import java.io.InputStream
+import java.io.OutputStream
 import java.nio.ByteBuffer
 import java.nio.file.Files
 import java.nio.file.Path
 import java.nio.file.Paths
 
 open class AvroInputStreamBuilder<T>(
-   private val converter: (Any) -> T
-   ) {
-   /**
-    * The format that should be used to decode the bytes from the input stream.
-    */
-   var decodeFormat : AvroDecodeFormat = defaultDecodeFormat
-   @Deprecated("please use decodeFormat to specify the format")
-   var format: AvroFormat = AvroFormat.DataFormat
-   @Deprecated("please use decodeFormat to specify the format")
-   var writerSchema: Schema? = null
-   @Deprecated("please use decodeFormat to specify the format")
-   var readerSchema: Schema? = null
-
-   companion object {
-      val defaultDecodeFormat = AvroDecodeFormat.Data(null, null)
-   }
-
-   private val derivedDecodeFormat : AvroDecodeFormat
-      get() {
-         return when(format){
-            is AvroFormat.JsonFormat -> {
-               val wschema = writerSchema ?: error("Writer schema needs to be supplied for Json format")
-               AvroDecodeFormat.Json(wschema, readerSchema?:wschema)
-            }
-            is AvroFormat.BinaryFormat -> {
-               val wschema = writerSchema ?: error("Writer schema needs to be supplied for Binary format")
-               AvroDecodeFormat.Binary(wschema, readerSchema?:wschema)
+    private val converter: (Any) -> T
```

Comment:
It is really hard to review this patch with all the reformating ...

Reply:
btw, i've put space formatting to 4 spaces, because a lot of files were using 4 spaces for tabs (but the Avro class only 3, it's a mess)